sidekick_plugin_installer/test/register_plugin_test.dart

```diff
@@ -0,0 +1,170 @@
+        );
+      });
+    });
+
+    test('A import gets added', () {
+      insideFakeProjectWithSidekick((dir) async {
+        dir.file('packages/dash/lib/dash.dart').writeAsStringSync(
+              initialCliFileContentWithoutImportAndCommand,
+            );
+        registerPlugin(
+          sidekickCli:
+              DartPackage.fromDirectory(dir.directory('packages/dash'))!,
+          import: "import 'package:my_package/src/my_command.dart';",
+          command: 'MyCommand()',
+        );
+        expect(
+          dir.file('packages/dash/lib/dash.dart').readAsStringSync(),
+          initialCliFileContentWithMyImport,
+        );
+      });
+    });
+    test('A import does not get double added if the command is already present',
+        () {
+      insideFakeProjectWithSidekick((dir) async {
+        dir.file('packages/dash/lib/dash.dart').writeAsStringSync(
+              initialCliFileContentWithMyImport,
+            );
+        registerPlugin(
+          sidekickCli:
+              DartPackage.fromDirectory(dir.directory('packages/dash'))!,
+          import: "import 'package:my_package/src/my_command.dart';",
+          command: 'MyCommand()',
+        );
+        expect(
+          dir.file('packages/dash/lib/dash.dart').readAsStringSync(),
+          initialCliFileContentWithMyImport,
+        );
+      });
+    });
+  });
+
+  group('throws error when arguments are not valid because', () {
+    final packageDir = Directory.systemTemp.createTempSync();
+    packageDir.file('pubspec.yaml')
+      ..createSync(recursive: true)
+      ..writeAsStringSync('''
+name: foo
+
+environment:
+  sdk: ^2.0.0
+''');
+    final package = DartPackage.fromDirectory(packageDir)!;
+    test('gitUrl is required but is missing', () {
+      expect(
+        () => addDependency(
+          package: package,
+          dependency: 'foo',
+          gitPath: 'bar',
+        ),
+        throwsA('git arguments were passed, but `gitUrl` was null.'),
+      );
+    });
+    test('too many arguments are given', () {
+      expect(
+        () => addDependency(
+          package: package,
+          dependency: 'foo',
+          gitPath: 'bar',
+          localPath: 'baz',
+        ),
+        throwsA(
+          'Too many arguments. Pass only one type of arguments (path/hosted/git).',
+        ),
+      );
+    });
+    packageDir.deleteSync(recursive: true);
+  });
+}
+
+const initialCliFileContentWithoutImportAndCommand = '''
+Future<void> runDash(List<String> args) async {
+  final runner = initializeSidekick(
+    name: 'dash',
+    flutterSdkPath: systemFlutterSdkPath(),
+  );
+  runner
+    ..addCommand(DartCommand())
+    ..addCommand(SidekickCommand());
+  try {
+    return await runner.run(args);
+  } on UsageException catch (e) {
+    print(e);
+    exit(64); // usage error
+  }
+}
+''';
+const initialCliFileContentWithMyImport = '''
+\nimport 'package:my_package/src/my_command.dart';Future<void> runDash(List<String> args) async {
```

Comment:
It's because we add one when we add an import:
https://github.com/phntmxyz/sidekick/blob/a8bd02510f1a9acbaca9e868a2e43cb0ffdf5cb1/sidekick_plugin_installer/lib/src/add_import.dart#L27

Do you want me to only add that newLine when there is no other import?
It get's formatted away afterwards though i think. Or run format before this step?

Reply:
only add `\n` when there are other imports above it

regardless if we run formatting afterwards. This code will most likely be used in other cases where we don't format.
Checking `position != 0` should be sufficient :)